apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/util/Util.java

```diff
@@ -2035,7 +2035,22 @@ public static File getEEDescriptionFile() {
 	 * @since 1.1
 	 */
 	public static File createTempFile(String prefix, String suffix) throws IOException {
-		File file = File.createTempFile(prefix, suffix);
+		File file = Files.createTempFile(prefix, suffix).toFile();
```

Comment:
In order to avoid code duplication I suggest to extract the shared code from the `createTempFile()` and createTempDirectory() method (in the GH UI I can only suggest changes for changed lines, not existing lines. Therefore you probably have to apply the suggestions manually).

Besides that javadoc parameters without any description are worthless and IMHO better removed (or a description is added, but I think that's not necessary here).

All of that (plus the removal of the  since-tag should be done in the first clean-up commit).


```suggestion
	/**
	 * Creates a new file in the users' <code>temp</code> directory
	 *
	 * @return a new temp file
	 * @throws IOException
	 * @since 1.1
	 */
	public static File createTempFile(String prefix, String suffix) throws IOException {
		return recordTempFile(Files.createTempFile(prefix, suffix).toFile());
	}

	/**
	 * Creates a new directory in the users' <code>temp</code> directory
	 *
	 * @return a new temp directory
	 * @throws IOException
	 */
	public static File createTempDirectory(String prefix) throws IOException {
		return recordTempFile(Files.createTempDirectory(prefix).toFile());
	}

	private static File recordTempFile(File file) throws IOException {
		file.deleteOnExit(); 
		FileManager.getManager().recordTempFileRoot(file.getCanonicalPath());
		return file;
	}
```